script.js

```diff
@@ -126,14 +137,18 @@ function initListeners() {
 
     if (target.classList.contains("options__trash")) {
       let updatedObj = notesObj.filter((note) => note.id !== noteId);
+
       notesObj = updatedObj;
+      saveNotesToLocalStorage();
     }
 
     // check note
     if (target.closest(".notes__item-check")) {
       notesObj = notesObj.map((note) =>
         note.id === noteId ? { ...note, isChecked: !note.isChecked } : note
       );
+
+      saveNotesToLocalStorage();
```

Comment:
Will be better to pass data to persist "notesObj" and also storage field name "notes" in arguments of the functions, this way the function will be testable and potentially reusable in other places